### ISSUE TITLE
Expense - Report title for p2p transaction is editable but the new title is not reflected

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#45183]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.6-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4706534
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to DM with any user.
3. Submit an expense to the user.
4. Click on the expense preview,
5. On expense report, click on the report header.
6. Repeat Step 2 to 5 with a new account if the issue is not reproduced (sometimes it does not happen with old accounts).

## Expected Result:
Expense report title for p2p transaction should not be editable.

## Actual Result:
Expense report title for p2p transaction is editable, and the edited title is not reflected.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/963e453c-844a-4838-a4cf-e36223b90960

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by etCoderDysto** [comment]
I think @ishpaul777 has a draft pr that fixes this issue.

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Expense report title for p2p transaction is editable, and the edited title is not reflected.

### What is the root cause of that problem?
For p2p transaction, the policyID of iou report or the transaction thread report is the personal policyID of the user. This policy has a title field that is enabled.

We only disable editing this field if the it's disabled
https://github.com/Expensify/App/blob/978b833cf10aa22f3f20257f645e31a2f64492b3/src/pages/ReportDetailsPage.tsx#L606

https://github.com/Expensify/App/blob/978b833cf10aa22f3f20257f645e31a2f64492b3/src/pages/ReportDetailsPage.tsx#L635
### What changes do you think we should make in order to solve the problem?
For p2p transaction, we don't need to show the title field. We can return here early if the policy is personal policy

```
if (policy?.type === CONST.POLICY.TYPE.PERSONAL) {
    return;
}
```

https://github.com/Expensify/App/blob/978b833cf10aa22f3f20257f645e31a2f64492b3/src/pages/ReportDetailsPage.tsx#L606

### What alternative solutions did you explore? (Optional)
Or if we want to disable this field we can add the check here to disable the title field if the policy is personal policy

```
shouldShowRightIcon={!isFieldDisabled && policy?.type !== CONST.POLICY.TYPE.PERSONAL}
interactive={!isFieldDisabled && policy?.type !== CONST.POLICY.TYPE.PERSONAL}
```
https://github.com/Expensify/App/blob/978b833cf10aa22f3f20257f645e31a2f64492b3/src/pages/ReportDetailsPage.tsx#L635

OPTIONAL: If we do this solution we can update the policyID of the report as personal policyID in optimistic data when we create a new transaction

**Comment by hungvu193** [comment]
I'll prioritize this one tomorrow morning.

**Comment by hungvu193** [comment]
@kadiealexander Do we need to enable the report title field for p2p transaction?

**Comment by kadiealexander** [comment]
What do you mean?

**Comment by hungvu193** [comment]
I mean, do we let user to edit their report title in ReportDetail page of a p2p transaction report?

**Comment by kadiealexander** [comment]
Yes, I believe so.

**Comment by hungvu193** [comment]
Cool. The I think we're still waiting for proposals

**Comment by nkdengineer** [comment]
@hungvu193 As the video in the OP when we edit the title it doesn't change so I think the user should not able to edit the title of p2p iou report.

**Comment by nkdengineer** [comment]
Additional for iou report the report name is always `IOU`.

**Comment by hungvu193** [comment]
I think this can be fixed from BE, IIRC I can't always reproduce this issue, I need to create a new account to reproduce it. I also think disable title field for p2p transaction is fine IMO. Let's me pull internal engineer here for the final decision.

🎀 👀 🎀

**Comment by hungvu193** [comment]
Hey @yuwenmemon, can you take a look on the convo above, ty 😄

**Comment by hungvu193** [comment]
little bump @yuwenmemon

**Comment by yuwenmemon** [comment]
Yes this sounds like a missing Onyx update.

**Comment by nkdengineer** [comment]
@yuwenmemon For the iou report, we don't use `reportName` to display in the report detail page so I think we should disable the report title section in `ReportDetailPage` of iou report.

**Comment by nkdengineer** [comment]
@yuwenmemon  As we can see in the video, the rename API is successful but we don't use `reportName` to display iou report. So I think it's not a BE bug.

cc @hungvu193 

https://github.com/user-attachments/assets/321c9fdb-3faf-4b25-a079-8ad258d3cc81

**Comment by nkdengineer** [comment]
@hungvu193 Any thoughts on this https://github.com/Expensify/App/issues/45183#issuecomment-2244180294?

**Comment by hungvu193** [comment]
I think this can be fixed from BE, I did submit 2 expenses to the same person, one with my old account, one with my brand new account.
With my old account I can't reproduce this issue, with my brand new account I can. The problem here is that, with brand new account our `report`'s `policy` has `filedList`, mean while, `fieldList` is undefined with old account.


<img width="951" alt="Screenshot 2024-07-25 at 13 45 13" src="https://github.com/user-attachments/assets/e1fd271d-cf29-4e3e-9fa3-f3d6272c2fc8">

https://github.com/user-attachments/assets/2bb08db6-0393-480c-bb3f-d8e1c5cff32f

**Comment by nkdengineer** [comment]
@yuwenmemon Can you please confirm if the personal policy of the new account should have the `filedList`?

**Comment by yuwenmemon** [comment]
No that's clearly a typo but I'm very confused. I don't see `filedList` in any of your screenshots? Where are you seeing this?

**Comment by yuwenmemon** [comment]
I think you guys are spelling this wrong haha. You mean `fieldList`? The original comment made it sound like there was some mismatch between `fieldList` and a typo `filedList`

**Comment by nkdengineer** [comment]
@yuwenmemon Yes it's `fieldList`, the personal policy of a new account now has `fieldList` with the `title` enabled.

<img width="1512" alt="Screenshot 2024-07-26 at 01 18 52" src="https://github.com/user-attachments/assets/70b95884-f1b8-4749-961f-c3ccd5a20c0c">

**Comment by yuwenmemon** [comment]
Hmmmm, looking at the backend locally, both policies (the new one with the issue and the old one without) have a `fieldList` defined in the DB. So something else is amiss 🤔

**Comment by yuwenmemon** [comment]
Ah, I think for new accounts the _active_ policy is your personal policy, and thus we end up loading the full data for it here rather than just the public info: https://github.com/Expensify/Auth/blob/6cc494432ba5b32520f95a02ba71af55ea99c2a9/auth/command/OpenApp.cpp#L249C1-L253C6

In any case I don't believe we need to send the `fieldList` for the personal policy ever, correct? @mountiny @thienlnam

**Comment by thienlnam** [comment]
Yes that's correct - personal policies should never have report fields

**Comment by hungvu193** [comment]
> No that's clearly a typo but I'm very confused. I don't see `filedList` in any of your screenshots? Where are you seeing this?

haha yeah, my typo 🤣 , should be `fieldList`

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by thienlnam** [comment]
This is done
